modules/standard/IO.chpl

```diff
@@ -3068,7 +3076,12 @@ private proc _write_text_internal(_channel_internal:qio_channel_ptr_t, x:?t):sys
   return EINVAL;
 }
 
+config param testReadBinaryInternalEIO = false;
```

Comment:
I see you've added a test for it specifically.  But is that useful?  It seems like you're not verifying something the function actually does otherwise, because that'd be harder to force.  But I don't think there's a scenario where a user would want to use this flag or get this explicit behavior.

Reply:
I put this flag in to force a low-level error during testing to make sure it gets converted into an exception and handled properly by the methods I wrote. Otherwise it's difficult to cause an I/O error during testing.

Reply:
I think if it doesn't get turned into an error and thrown, that's an issue with error handling rather than your function.  And in that case, it'd be reproducible in an independent test case rather than living in a module that users are expected to use.

Reply:
I think the flag is not generally useful but it's reasonable for testing. The problem it is solving is that some system errors are hard to trigger (e.g. the error you get when you run out of disk space, or your disk drive has a hardware error). I don't think a user would ever want to set it. You could prefix it with `chpl_` to indicate that. Since it is a `config param`, we don't currently show it in any help output or anything, so I don't think it's causing any harm to users. We have a variety of other `config param` s that are really only there for our own use.